apps/04TowersOfHanoi.js

```diff
@@ -18,51 +18,33 @@ function printStacks() {
     console.log("c: " + stacks.c);
 }
 
-function movePiece(startStack, endStack) {
-    var move = stacks[startStack].pop();
-    stacks[endStack].push(move);
-};
-//
-function isLegal(startStack, endStack){
-  if (stacks[endStack].length === 0){
+function movePiece (startStack, endStack) {
+  if (isLegal(startStack, endStack)=== true) {
+    stacks[endStack].push(stacks[startStack].pop());
+  }
+}
+
+function isLegal (startStack, endStack) {
+  if (stacks[endStack].length === 0) {
     return true;
-  }else{
-    if (startStack === 'a'){
-      start = stacks.a[stacks.a.length-1];
- 		}else if(endStack==='a'){
- 			end = stacks.a[stacks.a.length-1];
- 		}
- 		if (startStack === 'b'){
-			start = stacks.b[stacks.b.length-1];
- 		}else if(endStack==='b'){
- 			end = stacks.b[stacks.b.length-1];
- 		}
- 		if (startStack === 'c'){
-			start = stacks.c[stacks.c.length-1];
- 		}else if(endStack==='c'){
- 			end = stacks.c[stacks.c.length-1];
- 		}
-		if (start<end){
- 			return true;
- 		}else if (start>end){
- 			return false;
- 		}
- 	}
+  } else if (stacks[startStack][stacks[startStack].length-1] < stacks[endStack][stacks[endStack].length-1]) {
+    return true;
+  } else {
+    return false;
+  }
 }
 
-function checkForWin() {
-    if(stacks['b'].length === 4 || stacks['c'].length === 4 ){
+function checkForWin () {
+    if (stacks['b'].length === 4 || stacks['c'].length === 4 ) {
       console.log("You Won!!!");
       return true;
-    }else{
+    } else {
       return false;
     }
 }
 
-function towersOfHanoi(startStack, endStack) {
-    if (isLegal === true && checkForWin === false){
-      movePiece(startStack, endStack);
-    }
+function towersOfHanoi (startStack, endStack) {
+      movePiece (startStack, endStack);
```

Comment:
Great work going through all of this! Always assume that the user is going to make the worst possible mistakes.